source/MaterialXGenShader/TypeDesc.h

```diff
@@ -85,6 +93,9 @@ class MX_GENSHADER_API TypeDesc
     /// Return the semantic for the type.
     unsigned char getSemantic() const { return _semantic; }
 
+    /// Return the index for the struct member information in StructTypeDesc, the result is invalid if `isStruct()` returns false.
```

Comment:
Perhaps we should declare this method just below `isStruct`, so that the reference to `isStruct` in the comment is immediately clear to the reader.

Reply:
I thought it was most consistent to keep all the `getXXX()` functions together and all the `isXXX()` functions together.

If you feel strongly about moving the function so that the referenced function in the comment is close by then I'm ok with that,  just let me know.